src/core/qgsofflineediting.cpp

```diff
@@ -817,7 +831,7 @@ void QgsOfflineEditing::applyAttributeValueChanges( QgsVectorLayer* offlineLayer
   for ( int i = 0; i < values.size(); i++ )
   {
     QgsFeatureId fid = remoteFid( db, layerId, values.at( i ).fid );
-
+    QgsDebugMsg( QString( "Offline changeAttributeValue %1 = %2" ).arg( QString( attrLookup[ values.at( i ).attr ] ), values.at( i ).value ) );
```

Comment:
On 06/30/2016 09:55 AM, Alessandro Pasotti wrote:
> In src/core/qgsofflineediting.cpp
> <https://github.com/qgis/QGIS/pull/3258#discussion_r69086725>:
> 
>> @@ -817,7 +831,7 @@ void QgsOfflineEditing::applyAttributeValueChanges( QgsVectorLayer* offlineLayer
>>    for ( int i = 0; i < values.size(); i++ )
>>    {
>>      QgsFeatureId fid = remoteFid( db, layerId, values.at( i ).fid );
>> -
>> +    QgsDebugMsg( QString( "Offline changeAttributeValue %1 = %2" ).arg( QString( attrLookup[ values.at( i ).attr ] ), values.at( i ).value ) );
> 
> No problem to remove that debug message.
> Do you mean that the other debug messages should become
> |QgsDebugMsgLevel| ? What level do you think would be appropriate?

I always use 4 for stuff that I really only need for debugging.
Except for the ones that should actually never trigger (and depict real
errors) which don't need the level and could possibly go to
QgsMessageLog (and end up in the user interface)

> 
> I'm not sure that thing are really fixed for WFS, this test passes
> locally but I'd like to add another test for the case:
> 
>   * go offline
>   * edit
>   * delete the first or the last feature with an online WFS call
>   * sync
> 
> I fear that the fid<->id mapping will non be correct anymore, don't we
> really need a WFS fid<->id mapping stored in the offline DB?

Yeah, that would be a good idea.

I guess WFS fids are not stable, I don't know if there is a way to get
something unique and stable (pk) yet.